Sources/ArcGISToolkit/Components/AR/TableTopSceneView.swift

```diff
@@ -0,0 +1,241 @@
+            lastGoodDeviceOrientation = deviceOrientation
+        }
+    }
+    
+    func visualizePlane(_ renderer: SCNSceneRenderer, didAdd node: SCNNode, for anchor: ARAnchor) {
+        // Place content only for anchors found by plane detection.
+        guard let planeAnchor = anchor as? ARPlaneAnchor else { return }
+        
+        // Create a custom object to visualize the plane geometry and extent.
+        let plane = Plane(anchor: planeAnchor)
+        
+        // Add the visualization to the ARKit-managed node so that it tracks
+        // changes in the plane anchor as plane estimation continues.
+        node.addChildNode(plane)
+    }
+    
+    func updatePlane(_ renderer: SCNSceneRenderer, didUpdate node: SCNNode, for anchor: ARAnchor) {
+        if didSetTransforamtion {
+           node.removeFromParentNode()
+        }
+    
+        guard let planeAnchor = anchor as? ARPlaneAnchor,
+              let plane = node.childNodes.first as? Plane
+        else { return }
+        
+        // Update extent visualization to the anchor's new bounding rectangle.
+        if let extentGeometry = plane.node.geometry as? SCNPlane {
+            extentGeometry.width = CGFloat(planeAnchor.extent.x)
+            extentGeometry.height = CGFloat(planeAnchor.extent.z)
+            plane.node.simdPosition = planeAnchor.center
+        }
+    }
+}
+
+/// Helper class to visualize a plane found by ARKit
+class Plane: SCNNode {
+    let node: SCNNode
+    
+    init(anchor: ARPlaneAnchor) {
+        // Create a node to visualize the plane's bounding rectangle.
+        let extent = SCNPlane(width: CGFloat(anchor.extent.x), height: CGFloat(anchor.extent.z))
+        node = SCNNode(geometry: extent)
+        node.simdPosition = anchor.center
+        
+        // `SCNPlane` is vertically oriented in its local coordinate space, so
+        // rotate it to match the orientation of `ARPlaneAnchor`.
+        node.eulerAngles.x = -.pi / 2
+
+        super.init()
+
+        node.opacity = 0.25
+        guard let material = node.geometry?.firstMaterial
+            else { fatalError("SCNPlane always has one material") }
+        
+        material.diffuse.contents = UIColor.white
+
+        // Add the plane node as child node so they appear in the scene.
+        addChildNode(node)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+private extension SceneViewProxy {
+    /// Sets the initial transformation used to offset the originCamera.  The initial transformation is based on an AR point determined via existing plane hit detection from `screenPoint`.  If an AR point cannot be determined, this method will return `false`.
+    ///
+    /// - Parameter screenPoint: The screen point to determine the `initialTransformation` from.
+    /// - Returns: The `initialTransformation`.
+    /// - Since: 200.3
+    func setInitialTransformation(
```

Comment:
This doesn't set the initial transformation. It returns an initial transformation. Therefore it should be named as such. Maybe something like `initialTransformation(for:...)`

Also it doesn't seem to require the SceneViewProxy, so this method should extend SceneViewProxy. Maybe it should extend ARViewProxy.